xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/internal/reference/AbstractStringEntityReferenceResolver.java

```diff
@@ -115,6 +115,18 @@ private StringBuilder createStringBuilder(String entityReferenceRepresentation)
     @Override
     public EntityReference resolve(String entityReferenceRepresentation, EntityType type, Object... parameters)
     {
+        // If the type is not provided, try to extract it from the string
+        if (type == null) {
+            Character entityTypeSeparator = getSymbolScheme().getEntityTypeSeparator();
+            if (entityTypeSeparator != null) {
```

Comment:
No else branch?  no risk of NPE in the remaining of  the method if type remains null? AFAIR it wasn't meant to be nullable is it?

Reply:
Also not sure why you test for null here since you provided a default method in the interface: the overriding implementation should never return null for this.

Reply:
> No else branch? no risk of NPE in the remaining of the method if type remains null?

Probably, but then it will behave as it used to, I just added an improvement when and entity type separator is provided. The only thing that would make sense in a else is to throw a IllegalArgumentException.

> the overriding implementation should never return null for this.

Technically, nothing prevent an implementation to do `return null`.

Reply:
> The only thing that would make sense in a else is to throw a IllegalArgumentException.

Would be better I guess. 

> Technically, nothing prevent an implementation to do return null.

Sure in practice, but the javadoc should be clear that it's not supposed to be nullable IMO.

Reply:
> the javadoc should be clear that it's not supposed to be nullable IMO

Sure why not, but I still prefer to test this use case when it's technically possible :)